src/libcrun/utils.c

```diff
@@ -333,7 +333,8 @@ check_fd_under_path (const char *rootfs, size_t rootfslen, int fd, const char *f
   if (UNLIKELY (ret < 0))
     return crun_make_error (err, errno, "readlink `%s`", fdname);
 
-  if (((size_t) ret) <= rootfslen || memcmp (link, rootfs, rootfslen) != 0)
+  if (((size_t) ret) <= rootfslen || memcmp (link, rootfs, rootfslen) != 0
+      || (((size_t) ret) > rootfslen && link[rootfslen] != '/'))
```

Comment:
at this point we know `rootfslen > len` otherwise it would return an error, could we avoid the extra check of ret and simplify it as the following?

```suggestion
  if (((size_t) ret) <= rootfslen || memcmp (link, rootfs, rootfslen) != 0 || link[rootfslen] != '/'))
```

Reply:
yes, good point